### ISSUE TITLE
`QuantizationAwareTraining(on_fit_end=True)` breaks `Trainer.test(ckpt="path")`

### ISSUE THREAD

**Issue by ORippler** [Lightning-AI/lightning#12896]
## 🐛 Bug
When performing `QuantizationAwareTraining` with `quantize_on_fit_end=True`, the signature of calling `Trainer.test(ckpt= PATH)`
errors/no longer works.

### To Reproduce

https://colab.research.google.com/drive/1F2HJWkUSGWpoHgGLZgMaPiJMlZBU0AeM?usp=sharing

### Expected behavior

`Trainer.test(ckpt = PATH)` works normally, even for `QuantizationAwareTraining(quantize_on_fit_end=True)` (which is the current default)

### Environment

```
* CUDA:
	- GPU:
	- available:         False
	- version:           11.3
* Packages:
	- numpy:             1.21.6
	- pyTorch_debug:     False
	- pyTorch_version:   1.11.0+cu113
	- pytorch-lightning: 1.6.1
	- tqdm:              4.64.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.13
	- version:           #1 SMP Tue Dec 7 09:58:10 PST 2021
```

### Additional context

cc @awaelchli @ananthsub @ninginthecloud @rohitgr7 @otaj @borda @Borda

**Comment by ORippler** [comment]
Just reran the notebook, and it still errors out. Therefore pinging @Borda

**Comment by ORippler** [comment]
Notebook still throws an error, so apparently the issue is still present

**Comment by awaelchli** [comment]
Closing this stale issue because we removed the QAT callback in #16750. As explained in the linked PR:

The QAT callback can no longer be maintained by us. It has many issues that make the callback uneffective and these can't be fixed at the moment.

Users who rely on this callback: You can stay on the Lightning 1.9.x version which gets long-term support (LTS) OR you can copy the callback code and maintain it yourself.

If someone from the community is interested in fixing and maintaining this callback, please let us know. If important issues come up that affect several users, we will prioritize fixes accordingly for the 1.9.x LTS.
